server/ast/insert.go

```diff
@@ -93,10 +106,21 @@ func nodeInsert(ctx *Context, node *tree.Insert) (*vitess.Insert, error) {
 		With:    with,
 		Columns: columns,
 		Rows:    rows,
+		OnDup:   onDuplicate,
 		Auth: vitess.AuthInformation{
 			AuthType:    auth.AuthType_INSERT,
 			TargetType:  auth.AuthTargetType_SingleTableIdentifier,
 			TargetNames: []string{tableName.SchemaQualifier.String(), tableName.Name.String()},
 		},
 	}, nil
 }
+
+func supportedOnDuplicateKey(conflict *tree.OnConflict) bool {
```

Comment:
Can never go wrong with function comments. Also, is this really clearer than
```go
conflict.ArbiterPredicate == nil && conflict.Where == nil
```
especially since it's not being reused?